tests/bofire/strategies/test_enting.py

```diff
@@ -0,0 +1,153 @@
+import random
+
+import numpy as np
+import pytest
+from pydantic.utils import deep_update
+
+import bofire.data_models.strategies.api as data_models
+from bofire.benchmarks.api import Hartmann
+from bofire.data_models.features.api import Input
+from bofire.strategies.api import EntingStrategy
+from tests.bofire.strategies.test_base import domains
+
+VALID_ENTING_STRATEGY_SPEC = {
```

Comment:
This structure of testing for the strategies and its data models is deprecated. Just add valid and invalid ones in `tests/bofire/data_models/strategies.py`. Then they will be automatically tested for serialization and deserialization and for throwing the correct error. For the invalid ones: only test stuff which makes really sense and not basic pydantic functionality.

The import in `test_all.py` is then also obsolote. This has to be completely tiedied up.

Reply:
Hm, I think I might be misunderstanding this. The file you mentioned doesn't exist, and I'm modelling my tests after `tests/bofire/strategies/test_sobo.py` which still has the import in `test_all.py`. These tests also seem to be essentially testing basic pydantic functionality - could you please give an example where this isn't the case?

Could you also please explain how the automatic testing works? I think this will help me better understand how to write these tests, thank you.

Reply:
You are right, it is still used in `test_sobo` and so forth, but this needs to be tidied up. The correct link is `tests/bofire/data_models/specs/strategies.py`. There valid and invalid specs can be registered and the valid ones are then tested in `tests/bofire/data_models/serialization` for correct serialization and deserialization and the invalid ones are tested for invalidity in `tests/bofire/data_models/test_invalid.py`. This is achieved by using fixtures which are defined in `tests/bofire/conftest.py` (no change on the fixtures is necessary from your side, you only need to add at least one valid configureation in `tests/bofire/data_models/specs/strategies.py` ).

The plan is to make use of `add_invalid` only in scenarios where one has custom validators. An example would be for example here for the `TaskInput` as we are testing here that the custom validator works: https://github.com/experimental-design/bofire/blob/ccb0bc2998f35ede3118b490e0ce387244898305/tests/bofire/data_models/specs/features.py#L173

I think in your case no `add_invalid` is needed, so just add one valid configuration via `add_valid`. You have to provide there all values for the fields, also the ones which have default values.

Reply:
Thank you for explaining, this all makes sense.

I have implemented it, and have run `test_strategy_should_be_(de)serializable` and they're all working. This testing set up is very cool, I like it!